app/controllers/sessions_controller.rb

```diff
@@ -1,5 +1,16 @@
 class SessionsController < ApplicationController
+
   def new
+    if current_user
```

Comment:
We had to write something very similar for the login routing at first, it might be useful to make a helper method to handle these redirects in order to remove the duplication.